test_environment.js

```diff
@@ -11,23 +11,29 @@ class LocalTestEnvironment extends NodeEnvironment {
     }
 
     async setup() {
-        this.global.nearlib = require('near-api-js');
+        this.global.nearAPI = require('near-api-js');
+        this.global.nearlib = new Proxy(this.global.nearAPI, {
+            get(obj, prop) {
+                console.warn('[DEPRECATION WARNING] `nearlib` has been deprecated and will be removed from near-shell@WHICH?; use `nearAPI` instead');
```

Comment:
I think it's best to keep `nearlib` here, but issue deprecation for test_environment as a whole (i.e. change example tests to not depend on it).

I also think it makes sense to start using https://www.npmjs.com/package/depd for deprecation warnings as it has some UX improvements (like not spamming multiple times for same call-site).

Reply:
I like the idea of this, but before we deprecate the whole `test_environment` I want to be able to give people a clear upgrade path. In the deprecation message, we could link people to a PR in one of our example apps demonstrating how to remove the dependence on `test_environment`. 

I starting implementing such an upgrade in `token-contract-as`, but doing so is not straightforward yet. The simplest way to stop relying on `test_environment` is to copy a large chunk of it into your own app. I don't think we should rush removing `test_environment` until we have something better to recommend. I suspect that better way will be discovered as part of https://github.com/near-examples/counter/pull/44 

In the meantime, I think it's best to leave this file unchanged, so that we don't deprecate part of the `test_environment` interface in one version of `near-shell`, just to deprecate the entire thing a version or two later.

I've pushed an update with this change; please re-review the PR and leave a ✅ or ⛔️

Reply:
> In the meantime, I think it's best to leave this file unchanged, so that we don't deprecate part of the test_environment interface in one version of near-shell, just to deprecate the entire thing a version or two later.

sgtm